sdk/dotnet/Pulumi/Deployment/GrpcMonitor.cs

```diff
@@ -0,0 +1,39 @@
+using System.Threading.Tasks;
+using Grpc.Core;
+using Pulumirpc;
+
+namespace Pulumi
+{
+    class GrpcMonitor : IMonitor
+    {
+        private readonly ResourceMonitor.ResourceMonitorClient _client;
+        private readonly Engine.EngineClient _engine;
+
+        public GrpcMonitor(string engine, string monitor)
+        {
+            this._engine = new Engine.EngineClient(new Channel(engine, ChannelCredentials.Insecure));
+            this._client = new ResourceMonitor.ResourceMonitorClient(new Channel(monitor, ChannelCredentials.Insecure));
+        }
+
+        public async Task<InvokeResponse> InvokeAsync(InvokeRequest request)
+            => await this._client.InvokeAsync(request);
+
+        public async Task<ReadResourceResponse> ReadResourceAsync(Resource resource, ReadResourceRequest request)
+            => await this._client.ReadResourceAsync(request);
+
+        public async Task<RegisterResourceResponse> RegisterResourceAsync(Resource resource, RegisterResourceRequest request)
+            => await this._client.RegisterResourceAsync(request);
```

Comment:
you don't use the `Resource resource` in the body.  why was it necessary?

Reply:
i see.  you use it in tests.   don't love it.

Might consider that if we have our own IRpcDispatcher interface, we have a few classes/structs we define that are independent of grpc.  It could hold onto the data and the resource.  the Grpc impl would then create the grpc type, the test version could just check the data however it wants.

Not certain if this is worth it... but it's something i wouldn't mind us looking into.

Reply:
yeah, it's an ugly way to build up a list of resources for tests